crypto/asn1/d2i_pr.c

```diff
@@ -15,15 +15,70 @@
 #include <openssl/bn.h>
 #include <openssl/evp.h>
 #include <openssl/objects.h>
+#include <openssl/decoder.h>
 #include <openssl/engine.h>
 #include <openssl/x509.h>
 #include <openssl/asn1.h>
 #include "crypto/asn1.h"
 #include "crypto/evp.h"
+#include "internal/asn1.h"
 
-EVP_PKEY *d2i_PrivateKey_ex(int type, EVP_PKEY **a, const unsigned char **pp,
+static const char *pkey_type_to_encoder_name(int type)
```

Comment:
Another idea would be to have it in `p_lib.c`, alongside `evp_pkey_name2type()`...  `evp_pkey_type2name()`, maybe?
I'll leave that up to you, though, as this function isn't *currently* needed anywhere.  It just struck me that they would be the perfect matching pair...

Reply:
Sounds like a good idea to me..

Reply:
except they are not a 1-1 mapping, since one is for encoding.

Reply:
Not sure I understand what you're saying there.  To me, this function has every sign of a *generic* function that maps from EVP_PKEY types to names.  What am I missing?

Reply:
I am saying it is not 'exactly' the same mapping as name2type for a few items..
e.g TYPE_DHX => "DHX" if not the same as "DH X.92" => TYPE_DHX

Reply:
Yeahok.  I don't have a huge problem with that (and `name2type` could be made to accept both "DHX" and "DH X9.42")